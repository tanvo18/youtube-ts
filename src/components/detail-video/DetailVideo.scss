.detail-wrapper {
  padding: 10px 20px;

  .detail-description {
    display: flex;
    justify-content: space-between;
    align-items: center;
    border-bottom: solid 0.0625rem #eee;

    .detail-view {
      color: #757575;
    }

    .btn-rate {
      color: #8f8f8f;
      font-size: 20px;
      background: transparent;
    }
  }

  .channel-wrapper {
    padding: 20px 0;
    border-bottom: solid 0.0625rem #eee;

    &__title, &__time, &__description {
      font-size: 14px;
    }

    &__time {
      color: #c8c6c8;
    }
  }
}
