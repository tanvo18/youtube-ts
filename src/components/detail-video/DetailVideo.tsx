import React, { useState, useEffect } from 'react';
import { getVideosById } from '../../api/videoAPI';
import { convertFormatDate, roundNumber } from '../../utils/common';
import "./DetailVideo.scss";
import { IVideoItem } from '../../interfaces/Interface';

interface IProps {
  videoId: string
}

const DetailVideo: React.FC<IProps> = ({ videoId }) => {
  const videoUrl = `http://www.youtube.com/embed/${videoId}`;
  const defaultVideo = {
    snippet: {
      title: '',
      channelTitle: '',
      publishedAt: '',
      description: ''
    },
    statistics: {
      viewCount: '',
      likeCount: '',
      dislikeCount: '',
    }
  }
  const [selectedVideo, setVideo] = useState<IVideoItem | any>(defaultVideo);

  useEffect(() => {
    // Using api to fetch videos 
    getVideosById(videoId)
      .then((response: any) => {
        let videoItem = response.data.items[0];
        videoItem.snippet.publishedAt = convertFormatDate(videoItem.snippet.publishedAt);
        videoItem.statistics.viewCount = roundNumber(videoItem.statistics.viewCount);
        videoItem.statistics.likeCount = roundNumber(videoItem.statistics.likeCount);
        videoItem.statistics.dislikeCount = roundNumber(videoItem.statistics.dislikeCount);

        setVideo(videoItem);
      })
  }, []);

  return (
    <div className="row detail-section">
      <div className="col-8">
        <div className="detail-wrapper">
          <iframe src={videoUrl}
            title={'title'}
            width="100%"
            height="580"
            frameBorder="0"
            allowFullScreen>
          </iframe>

          <div>
            <p className="detail-title">{selectedVideo.snippet.title}</p>
            <div className="detail-description">
              <p className="detail-view">{selectedVideo.statistics.viewCount} views</p>
              <div>
                <button className="btn btn-rate"><i className="fa fa-thumbs-up"></i></button>
                <span>{selectedVideo.statistics.likeCount}</span>
                <button className="btn btn-rate"><i className="fa fa-thumbs-down"></i></button>
                <span>{selectedVideo.statistics.dislikeCount}</span>
              </div>
            </div>

            <div className="channel-wrapper">
              <p className="channel-wrapper__title">{selectedVideo.snippet.channelTitle}</p>
              <p className="channel-wrapper__time">Published on {selectedVideo.snippet.publishedAt}</p>
              <p className="channel-wrapper__description">{selectedVideo.snippet.description}</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default DetailVideo
