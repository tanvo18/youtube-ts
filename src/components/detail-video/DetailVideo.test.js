import React from 'react';
import DetailVideo from './DetailVideo';
import renderer from 'react-test-renderer';

const props = {
  videoId: 'gIOyB9ZXn8s'
};

describe('test DetailVideo', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<DetailVideo {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
