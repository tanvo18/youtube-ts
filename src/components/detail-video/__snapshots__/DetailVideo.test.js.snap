// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DetailVideo should render correctly 1`] = `
<div
  className="row detail-section"
>
  <div
    className="col-8"
  >
    <div
      className="detail-wrapper"
    >
      <iframe
        allowFullScreen={true}
        frameBorder="0"
        height="580"
        src="http://www.youtube.com/embed/gIOyB9ZXn8s"
        title="title"
        width="100%"
      />
      <div>
        <p
          className="detail-title"
        >
          
        </p>
        <div
          className="detail-description"
        >
          <p
            className="detail-view"
          >
            
             views
          </p>
          <div>
            <button
              className="btn btn-rate"
            >
              <i
                className="fa fa-thumbs-up"
              />
            </button>
            <span>
              
            </span>
            <button
              className="btn btn-rate"
            >
              <i
                className="fa fa-thumbs-down"
              />
            </button>
            <span>
              
            </span>
          </div>
        </div>
        <div
          className="channel-wrapper"
        >
          <p
            className="channel-wrapper__title"
          >
            
          </p>
          <p
            className="channel-wrapper__time"
          >
            Published on 
            
          </p>
          <p
            className="channel-wrapper__description"
          >
            
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
